source/loader/layers/mock/ur_mock_layer.hpp

```diff
@@ -0,0 +1,79 @@
+/*
+ *
+ * Copyright (C) 2023-2024 Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ * See LICENSE.TXT
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * @file ur_layer.h
+ *
+ */
+#pragma once
+#include "ur_ddi.h"
+#include "ur_proxy_layer.hpp"
+#include "ur_util.hpp"
+
+#include <atomic>
+
+namespace ur_mock_layer {
+
+struct dummy_handle_t_ {
```

Comment:
I haven't hooked up the storage and data members of this yet, I'm kind of hoping that when it comes to porting the particular sycl unittests that were using them I can achieve the same effect with callbacks defined in the test rather than default behavior in our layer. I've left them in for now in case that isn't so.